haiku_node/client.py

```diff
@@ -52,6 +55,19 @@ def make_data_request(
         Make a data request from one App to another.
         """
 
+        # Check if the providing app is valid according to MOTHER
+        conf = UnificationConfig()
+        eos_client = Client(
+            nodes=[f"http://{conf['eos_rpc_ip']}:{conf['eos_rpc_port']}"])
+
+        v = UnificationAppScValidation(
+            eos_client, conf['acl_contract'],  providing_app.name,
+            get_perms=True)
+
+        if not v.valid():
```

Comment:
Do we need to check if an app is invalid both in the client and the rpc server?

Reply:
this is checking whether or not the providing app is valid, so should be done from the requesting app